bootstrap_cfn/fab_tasks.py

```diff
@@ -393,39 +396,70 @@ def set_stack_name():
     dns records to retreive it in the future.
 
     """
+    # create a stack id
+    r53_conn = get_connection(R53)
+    zone_name = get_zone_name()
+    zone_id = get_zone_id()
+    stack_suffix = uuid.uuid4().__str__()[-8:]
     if hasattr(env, 'tag'):
-        stack_tag = env.tag
+        if env.tag == 'active':
+            raise ActiveTagExistConflictError(stack_suffix)
+        elif r53_conn.hastag(zone_name, zone_id, get_tag_record_name(env.tag)):
+            raise TagRecordExistConflictError(env.tag)
+        else:
+            stack_tag = get_env_tag()
     else:
-        stack_tag = 'active'
+        stack_tag = stack_suffix
         env.tag = stack_tag
-    legacy_name = "{0}-{1}".format(env.application, env.environment)
-    # get_config needs a stack_name so this is a hack because we don't
-    # know it yet...
-    env.stack_name = 'temp'
-    cfn_config = get_config()
-    r53_conn = get_connection(R53)
-    zone_name = cfn_config.data.get('master_zone', None)
+    record = "{}.{}".format(get_tag_record_name(stack_tag), zone_name)
+    logger.info("fab_tasks::set_stack_name: "
```

Comment:
`format` is better than using `%s` in formatting a string, however, in `logging`, `%s` is actually preferred, so you would do:
```
logger.info("Creating stack suffix '%s' for record '%s'", stack_suffix, record)
```